roles/openshift_health_checker/openshift_checks/port_availability.py

```diff
@@ -0,0 +1,48 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class PortAvailability(OpenShiftCheck):
+    """Check that required ports are not in use by another process."""
+
+    name = "port_availability"
+    tags = ["preflight"]
+
+    @classmethod
+    def is_active(cls, task_vars):
+        return super(PortAvailability, cls).is_active(task_vars)
+
+    def run(self, tmp, task_vars):
+        required_ports = [{
+            "port_name": "dnsmasq",
+            "port": 53,
+        }]
+
+        args = {
+            "ports": required_ports,
+        }
+
+        check_ports = self.execute_module("portavail", args, task_vars)
+        if check_ports["failed"]:
+            failed_ports = check_ports.get("error_ports", [])
+            msg = ("The following ports are required for a successful OpenShift\n"
+                   "installation, but are currently in use by other processes:\n\n{}\n\n{}")
+
+            msg = msg.format(
+                "\n".join(
+                    [
+                        '  - {} (required by "{}")'.format(
+                            str(p.get("port", "n/a")),
+                            p.get("port_name", "n/a")
+                        )
+                        for p in failed_ports
+                    ]
+                ),
+                ('Please refer to the following article for more information\n'
+                 'on required OpenShift ports:\n\n'
+                 'https://access.redhat.com/solutions/45294')
```

Comment:
@sosiouxme was not sure if I should link a customer portal article, or provide a command line solution to find the process currently using a port. Also, found this alternative customer portal article that may be relevant as well: https://access.redhat.com/solutions/1520653